Code/Framework/AzCore/AzCore/Memory/MemoryMarker.h

```diff
@@ -48,15 +58,42 @@ namespace AZ
         MemoryTagMarker(MemoryTagValue v);
         ~MemoryTagMarker();
     };
-}
+
+    class AssetMemoryTagMarker
+    {
+    public:
+        AssetMemoryTagMarker(const char* name);
+        ~AssetMemoryTagMarker();
+    };
+
+    class AssetMemoryTagLimit
+    {
+    public:
+        AssetMemoryTagLimit(size_t limit);
+        ~AssetMemoryTagLimit();
+
+        static void SetLimit(size_t limit);
+        static size_t GetLimit();
+    };
+
+} // namespace AZ
 
 #define MEMORY_ALLOCATION_MARKER AZ::MemoryAllocationMarker(nullptr, __FILE__, __LINE__)
 
+// without these wrapper macros Name##_LINE__ is Name__LINE__ but not Name123
+#define CONCATENATION_MACRO_1(a, b) a##b
+#define CONCATENATION_MACRO_2(a, b) CONCATENATION_MACRO_1(a, b)
+
 // the name must be a string literal
-#define MEMORY_ALLOCATION_MARKER_NAME(name) AZ::MemoryAllocationMarker(name, __FILE__, __LINE__)
+#define MEMORY_ALLOCATION_MARKER_NAME(name) AZ::MemoryAllocationMarker CONCATENATION_MACRO_2(memoryNameMarker_, __LINE__)(name, __FILE__, __LINE__)
+
+#define MEMORY_TAG(x) AZ::MemoryTagMarker CONCATENATION_MACRO_2(memoryTagMarker_, __LINE__)(AZ::MemoryTagValue::x)
+#define MEMORY_TAG_GAME(x) AZ::MemoryTagMarker CONCATENATION_MACRO_2(memoryTagMarkerGame_, __LINE__)(static_cast<AZ::MemoryTagValue>((unsigned int)(x)))
 
-#define MEMORY_TAG(x) AZ::MemoryTagMarker tagMarker##__LINE__(AZ::MemoryTagValue::x)
-#define MEMORY_TAG_GAME(x) AZ::MemoryTagMarker tagMarker##__LINE__(static_cast<AZ::MemoryTagValue>((unsigned int)(x)))
+#define ASSET_TAG(x) AZ::AssetMemoryTagMarker CONCATENATION_MACRO_2(assetMemoryTagMarker_, __LINE__)(x)
+#define MEMORY_ASSET_LIMIT(x) AZ::AssetMemoryTagLimit CONCATENATION_MACRO_2(assetMemoryTagLimit_, __LINE__)(x)
+#define MEMORY_SET_ASSET_LIMIT(x) AZ::AssetMemoryTagLimit::SetLimit(x)
+#define MEMORY_GET_ASSET_LIMIT() AZ::AssetMemoryTagLimit::GetLimit()
```

Comment:
Sometimes I cannot use an object that sets the limit in constructor, for example if it is conditional. I also need get/set functions.